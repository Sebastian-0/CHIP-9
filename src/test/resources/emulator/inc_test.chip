
# Test simple incs
ldi A, 1
ldi B, 1
ldi C, 2
ldi D, 3
ldi E, 4
ldi H, 5
ldi L, 255

inc A
inc B
inc C
inc D
inc E
inc H
inc L

sout A
sout B
sout C
sout D
sout E
sout H
sout L

ldx HL, 0x1010
ldi (HL), 7
inc (HL)

sout (HL)

# Test flags
ldi A, 15
ldi B, 255
ldi C, 1
ldi D, 253

# Half-Carry
inc A
sout FLAGS

# Carry, Half-Carry and Zero flags
inc B
sout FLAGS

# No flags
inc C
sout FLAGS

# Negative flag
inc D
sout FLAGS

hcf