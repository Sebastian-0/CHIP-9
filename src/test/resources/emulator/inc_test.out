2
2
3
4
5
6
0
8

32
176
0
64
